### ISSUE TITLE
Don't know what to do with a #<Asciidoctor::Table:0x1584c019>

### ISSUE THREAD

**Issue by alainsahli** [asciidoctor/asciidoctorj#472]
Hi

I wrote some simple code where I convert an asciidoc containing a table using AsciidoctorJ:

```java
Asciidoctor asciidoctor = Asciidoctor.Factory.create();
String adocContent = IOUtils.readFull(new FileInputStream("test.adoc")); // content is below
String convertedContent = asciidoctor.convert(adocContent, OptionsBuilder.options().backend("html5"));  // exception is thrown
```

I always get this error: `java.lang.IllegalArgumentException: Don't know what to do with a #<Asciidoctor::Table:0x1584c019>`. After debugging I found out that the problem seems to come from the `NodeConverter` in the method `createASTNode` where `Asciidoctor::Table` does not match any case.

The content of my test asciidoc file is the following:

```adoc
= Main title

|===
|Col1 |Col2

|Row1 Col1
|Row1 Col2

|Row2 Col1
|Row2 Col2
```

If I convert the same file using `asciidoctor` on the command line with the html5 backend it works. Is this a bug or did I do something wrong?

**Comment by alainsahli** [comment]
... I'm using asciidoctorj version 1.5.3.2

**Comment by robertpanzer** [comment]
Hi Alain,

could you please retry with AsciidoctorJ 1.5.4?
I tried your example with the current master and could not reproduce the problem.

And NodeConverter should also be called if you use extensions, could it be that you also have an extension registered somehow?

It would help a lot if you could also send full stack traces instead of only the message.

Cheers
Robert

**Comment by alainsahli** [comment]
I tried with version 1.5.4 but the same error happens:

```
java.lang.IllegalArgumentException: Don't know what to do with a #<Asciidoctor::Table:0x47ac613b>

	at org.asciidoctor.ast.NodeConverter.createASTNode(NodeConverter.java:54)
	at org.asciidoctor.ast.AbstractBlockImpl.getBlocks(AbstractBlockImpl.java:60)
	at org.asciidoctor.ast.AbstractBlockImpl.blocks(AbstractBlockImpl.java:49)
	at org.asciidoctor.internal.JRubyAsciidoctor.toDocument(JRubyAsciidoctor.java:184)
	at org.asciidoctor.internal.JRubyAsciidoctor.readDocumentStructure(JRubyAsciidoctor.java:246)
	at org.sahli.asciidoc.confluence.publisher.converter.AsciidocConfluencePage.structuredDocument(AsciidocConfluencePage.java:90)
	at org.sahli.asciidoc.confluence.publisher.converter.AsciidocConfluencePage.newAsciidocConfluencePage(AsciidocConfluencePage.java:72)
	at org.sahli.asciidoc.confluence.publisher.converter.AsciidocConfluencePageTest.table_withHeaderAndMultipleColumns_returnsConfluencePageWithContentHavingTableInCorrectMarkup(AsciidocConfluencePageTest.java:243)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:119)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:42)
	at com.intellij.rt.execution.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:234)
	at com.intellij.rt.execution.junit.JUnitStarter.main(JUnitStarter.java:74)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.intellij.rt.execution.application.AppMain.main(AppMain.java:144)
```

You can have a look to my code here: https://github.com/alainsahli/confluence-publisher/blob/bugfix/asciidoctorj-table/asciidoc-confluence-publisher-converter/src/test/java/org/sahli/asciidoc/confluence/publisher/converter/AsciidocConfluencePageTest.java#L230

And the implementation: https://github.com/alainsahli/confluence-publisher/blob/bugfix/asciidoctorj-table/asciidoc-confluence-publisher-converter/src/main/java/org/sahli/asciidoc/confluence/publisher/converter/AsciidocConfluencePage.java#L86

**Comment by abelsromero** [comment]
The issue is not while converting but while parsing the document with the `readDocumentStructure`.
As @alainsahli indicated the `createASTNode' method does not deal with this type of node...but good news is that it's already solved in the [1.6.0 branch](https://github.com/asciidoctor/asciidoctorj/blob/asciidoctorj-1.6.0/asciidoctorj-core/src/main/java/org/asciidoctor/ast/NodeConverter.java#L170).

I tested the branch `bugfix` in your project with `1.6.0-alpha.3` version and all tests worked. However, vesion 1.6.0 has no oficial scheduled version yet and is susceptible to some changes.
